fetch.bs

```diff
@@ -2172,6 +2264,216 @@ exploits pertain to those <a for=request>types</a>. Also, considering "<code>ima
+to the target <a for=request>origin</a>. The user agent does this while processing the
+`<code>Include-Referred-Token-Binding-ID</code>` <a for=response>header</a> in the redirect
+<a for=/>response</a>, by adding a <a for=/>referred-token-binding ID</a> in the
+<a for=/>Token Binding Message</a> that is created when the <a for=/>request</a>
+is sent to the target server.
+
+<h4 id=getting-token-binding-key>Getting a Token Binding Key</h4>
+
+<p>Get the <a for=/>token-binding key</a> for an <a for=/>origin</a> <var>tokenBindingOrigin</var>
+and <a for=/>token-binding key parameters</a> <var>tokenBindingKeyParameters</var>,
+using the user agent's <a for=/>token-binding key</a> store, by running these substeps:
+
+<ol>
+ <li><p>Let <var>keyDomainName</var> be null.
+
+ <li><p>If <var>tokenBindingOrigin</var>'s host is an IP address, set <var>keyDomainName</var> to
+ <var>tokenBindingOrigin</var>'s host.
+
+ <p>Otherwise, set <var>keyDomainName</var> to the registrable domain of
+ <var>tokenBindingOrigin</var>'s host, determined by running the steps in the Algorithm section of
+ <a href=https://publicsuffix.org/list/>Public Suffix List</a>
+ [[PUBLIC-SUFFIX]]
+
+ <li><p>Let <var>tokenBindingKeyPair</var> be null.
+
+ <li><p>Set <var>tokenBindingKeyPair</var> to the result of looking up the
+ <a for=/>token-binding key</a> pair for <var>keyDomainName</var> in the
+ <a for=/>token-binding key</a> store.
+
+ <li><p>If <var>tokenBindingKeyPair</var> is null, create a new key pair for
+ <var>keyDomainName</var> using <var>tokenBindingKeyParameters</var>, save it in the
+ <a for=/>token-binding key</a> store, and set <var>tokenBindingKeyPair</var> to that new key pair.
+
+ <li><p>Return <var>tokenBindingKeyPair</var>.
+</ol>
+
+
+<h4 id=computing-token-binding-header-value>Computing a Token Binding Header Value</h4>
+
+<p>If the user agent supports <a for=/>Token Binding</a>, build the value that can be passed
+in a `<code>Sec-Token-Binding</code>` <a for=/>header</a> for a <a for=/>request</a>
+<var>httpRequest</var> and a <a>connection</a> <var>tlsConnection</var>,
+using the <a for=/>token-binding key</a> store and the
+<a for=connection>connection pool</a>, by running these substeps:
+
+<ol>
+ <li><p>If <var>httpRequest</var>'s <a for=request>use-token-binding flag</a> is not set,
+ then return null.
+
+ <li><p>Initialize <var>tokenBindingMessage</var> to null.
+
+ <li><p>If Token Binding was not negotiated for <var>tlsConnection</var>, return null.
+
+ <li><p>Let <var>providedTokenBindingKeyPair</var> be the result of getting the
+ <a for=/>token-binding key</a> for the <a for=/>origin</a> of <var>httpRequest</var>'s
+ <a for=request>current url</a> and <var>tlsConnection</var>'s
+ <a for=/>token-binding key parameters</a> metadata, obtained by following the steps in the
+ <a href=#getting-token-binding-key>Getting a Token Binding Key</a> section of this document.
```

Comment:
As above, this step can explicitly call out to the "get the token-binding key for an origin" algorithm, which you can do once it's a real `<dfn>`. Then you won't need this indirection of "follow the steps over here".